### PR TITLE
[MRG] HW2

### PR DESCRIPTION
**Pull request by ngreenwald89**
findPrimeFactor throws NPE if a or b is null. Modified Contract to explain that. Added tests.

### CODE REVIEW

File: hw2/src/findPrimeFactor/FindPrimeFactor.java [hunk 1]
```diff
@@ -21,17 +20,15 @@ public static int findPrimeFactor (List<Integer> a, List<Integer> b) throws Null
         // Also note that a[] and b[] need not be of the same length.
 
 
-        for (int i = 0; i <= b.size(); i++) {
+        for (int i = 0; i < b.size(); i++) {
```
Comment (kjacks21) [inline hw2/src/findPrimeFactor/FindPrimeFactor.java]: I think this a bug that wasn't caught by the grader but was caught by `result8`.

**Comment by kjacks21** [comment]
@1000DS000000 just in case, if you want to pull this branch on work on it locally, do the following:

```
$ git fetch origin hw2-nat:hw2-nat
$ git checkout hw2-nat
```

and after adding and commiting (let me know if you want some guidance on how to do these two things) your work, 

```
$ git push origin hw2-nat
```

will push your changes to this branch and PR. Thanks!

**Comment by kjacks21** [comment]
What's all the .html stuff for? Also, I'm reconsidering throwing NPE for the first two preconditions, where we should actually throw an `IllegalArgumentException`. See this [SO](https://stackoverflow.com/questions/3881/illegalargumentexception-or-nullpointerexception-for-a-null-parameter) discussion.

**Comment by 1000DS000000** [comment]
eclipse has a generate Javadoc button that creates the javadoc like the online version. I don't know if that is what is being asked for or just the javadoc comments in the code.

**Comment by kjacks21** [comment]
> As for the exceptions i think we should stick to what it says in the book, where IllegalArgument is for non-null parameters and NPE for null parameters

Sounds good to me. Also, I posted on piazza to get more clarification on the javaDoc.

File: hw2/src/findPrimeFactor/FindPrimeFactor.java [hunk 2]
```diff
@@ -31,6 +31,11 @@ public static int findPrimeFactor (List<Integer> a, List<Integer> b)
         // Also note that a[] and b[] need not be of the same length.
 
 
+        //check empty list
```
Comment (kjacks21) [inline hw2/src/findPrimeFactor/FindPrimeFactor.java]: I added additional tests where list a is empty and was thrown and `IndexOutOfBoundsException`. Checking for empty lists resolved it. Are you guys satisfied with this?

**Comment by kjacks21** [comment]
@ngreenwald89 @1000DS000000 this should do it. I'm satisfied with the javaDocs and it isn't clear if we need both the docstring above and below the method declaration. Unless people have objections I'd just keep both. Please review `contributions.txt`. Also, please review the additional checking for empty lists (I left a comment above). Once people are good, I'll submit. Thanks!

**Comment by 1000DS000000** [comment]
I think its good to go, and i agree about leaving out the .htmls.

**Pull request merged**